From 3e8e83cefa30fe0f1f64dc8c13b6089ed8629df8 Mon Sep 17 00:00:00 2001
From: Nikolay Nechaev <Nikolay_Nechaev@mail.ru>
Date: Sat, 13 May 2023 13:31:19 +0300
Subject: [PATCH 2/2] Make telegram-desktop undumpable (via `prctl`)

Temporary solution to a bug in telegram-desktop: prevent it from being
core-dumped on failures.
---
 Telegram/SourceFiles/main.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Telegram/SourceFiles/main.cpp b/Telegram/SourceFiles/main.cpp
index d3d938c..83bd652 100644
--- a/Telegram/SourceFiles/main.cpp
+++ b/Telegram/SourceFiles/main.cpp
@@ -6,8 +6,13 @@ For license and copyright information please follow this link:
 https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 */
 #include "core/launcher.h"
+#include <sys/prctl.h>
+#include <linux/prctl.h>
 
 int main(int argc, char *argv[]) {
+    if (0 != prctl(PR_SET_DUMPABLE, 0, 0, 0, 0))
+        return 21;
+
 	const auto launcher = Core::Launcher::Create(argc, argv);
 	return launcher ? launcher->exec() : 1;
 }
-- 
2.40.1

